lib/tests-from-pull-request.js

```diff
@@ -1,32 +1,32 @@
-const fs = require('fs')
-const readline = require('readline')
+module.exports = getTestsFromPR;
 
-module.exports = getTestsFromPR
+function getTestsFromPR(context) {
+  const pull = context.issue();
+  const {pullRequests} = context.github;
 
-function getTestsFromPR (context) {
   return Promise.resolve()
-    .then(() => ['Hi Or'])
-  // .then(() => _getTestsFromPR(context))
+    .then(pullRequests.getFiles(pull))
+    .then((res) => res.data)
+    .then((files) => files.map((file) => file.patch))
+    .then((patches) => patches.join('\n'))
+    .then(_prPatch);
 }
 
-function _getTestsFromPR (context) {
-  const pull = context.issue()
-  const {repos, pullRequests} = context.github
-  // todo: code to get file contect
-  let newTestLines = []
-  let PRfiles
-  return pullRequests.getFiles(pull)
-    .then((res) => res.data)
-    .then((files) => PRfiles = files)
-    .then(() => console.log('FILES FROM PR:\n', PRfiles))
+function _prPatch(prPatch) {
+  const newTestLines = [];
+  const parsedPatch = prPatch.split('\n');
+  parsedPatch.forEach((line) => {
+    if (line.startsWith('+')) {
+      line = line.slice(1).trim(); // remove the + and spaces
+      if (line.startsWith('it(')) {
+        const _getTestTextFromPatchRegex = /(?<=\(').+?(?='\))/;
```

Comment:
regex 101 says otherwise

Reply:
It is a valid regex, but it's over engineered for this job so we need to change this anyway.

Also, it doesn't match all cases. For example:
- `it('should do it', () => {`
- `it('should do that', predefinedFunc)`
- `it('should it', function() {`

Are not matched by this regex.
So maybe hound did smell something here :-)

Reply:
Here's a regex I think is good enough: `/it\((.*),/`

it will capture everything between `it(` and a comma **in a greedy way**.

So, for example:

- `it('should do it', () => {`
  > capture group result: `'should do it'`
- it('should do that', predefinedFunc)
  > capture group result: `'should do that'`
- it('should it', function() {
  > capture group result: `'should it'`
- it('should do it' , () => {
  > capture group result: `'should do it' `
  > with the whitespace at the **end**
- it( 'should do that', predefinedFunc)
  > capture group result: ` 'should do that'`
  > with the whitespace at the **start**
- it( 'should it' , function() {
  > capture group result: ` 'should it' `
  > with the whitespaces before and after
- it( 'title with a , comma' , function() {
  > capture group result: ` 'title with a , comma' `
  > The reason it will also catch the comma in the title itself is that we used `.*` instead of `.*?`. The difference is that the 1st one is greedy and will **stop at the last match**, and the second one tries to match as few times as possible, so it will stop on the first comma (inside the title)

Because it can also include trailing whitespace, we should **trim** the matched string and slice the quotes out of it:
```javascript
// 1 to remove the first character and -1 to make
// it stop **1 character from the end**
`   'title with a , comma' `.trim().slice(1,-1);
```